src/snowflake/snowpark/session.py

```diff
@@ -817,19 +825,29 @@ def clear_packages(self) -> None:
         """
         self._packages.clear()
 
-    def add_requirements(self, file_path: str, force_push: bool = True) -> None:
+    def add_requirements(
+        self,
+        file_path: str,
+        force_push: bool = False,
+        force_install: bool = False,
+        persist_path: str = None,
+    ) -> None:
         """
         Adds a `requirement file <https://pip.pypa.io/en/stable/user_guide/#requirements-files>`_
         that contains a list of packages as dependencies of a user-defined function (UDF). Pure Python packages that
         are not available in Snowflake will be pip installed locally and made available as an import (via zip file
-        on a remote stage).
+        on a remote stage). You can specify the remote stage as `persist_path` to create a persistent environment.
 
          Note that this function also supports addition of requirements via a `conda environment yaml file
          <https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#create-env-file-manually>_`.
 
         Args:
             file_path: The path of a local requirement file.
             force_push: Force upload Python packages with native dependencies.
+            persist_path: A remote stage directory path where packages not present in Snowflake will be persisted. Mentioning
+             this path will speed up automated package loading.
+            force_install: Ignores environment present on persist_path and overwrites it with a fresh installation.
+
```

Comment:
force_install is useful for "cleaning" environments from your persistent path

Reply:
does it mean that if hash generated by package versions is same but the zip file changes, the old zip file will be overwritten?

Reply:
It means that there will be a new zip file pushed to the stage regardless of whether the hash generated is same.